.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,128 @@
+    - name: Install Plugin into OpenSearch for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest'}}
+      run: |
+        cat > os-ep.sh <<EOF
+        yes | opensearch-plugin install file:///docker-host/${{ inputs.plugin-zip }}.zip
+        chmod +x plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        yes | plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        chown 1001:1001 -R /opensearch
+        su -c "/opensearch/bin/opensearch" -s /bin/bash opensearch
+        EOF
+        docker build -t opensearch-test:latest -f- . <<EOF
+        FROM ubuntu:latest
+        COPY --chown=1001:1001 os-ep.sh /docker-host/
+        COPY --chown=1001:1001 ${{ inputs.plugin-name }}.zip /docker-host/${{ inputs.plugin-zip }}.zip
+        COPY --chown=1001:1001 opensearch* /opensearch/
+        RUN chmod +x /docker-host/os-ep.sh
+        RUN useradd -u 1001 -s /sbin/nologin opensearch
+        ENV PATH="/opensearch/bin:${PATH}"
+        WORKDIR /opensearch/
+        ENTRYPOINT /docker-host/os-ep.sh
+        EOF
+      shell: bash
+
+    - name: Install Plugin into OpenSearch for Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: |
+        'y' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch-plugin.bat install file:$pwd\${{ inputs.plugin-name }}.zip
+        'y', 'y', 'N' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\plugins\${{ inputs.plugin-name }}\tools\${{ inputs.plugin-start-script }}.bat
+      shell: pwsh
+
+    # Run OpenSearch
+    - name: Run OpenSearch with plugin on Linux
+      if: ${{ inputs.platform == 'ubuntu-latest'}}
+      run: docker run --name ops -d -p 9200:9200 -p 9600:9600 -i opensearch-test:latest
+      shell: bash
+
+    - name: Run OpenSearch with plugin on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: start .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch.bat
+      shell: pwsh
+
+    # Give the OpenSearch process some time to boot up before sending any requires, might need to increase the default time!
+    - name: Sleep while OpenSearch starts
+      uses: peternied/action-sleep@v1
+      with:
+        seconds: 30
+
+    # Verify that the server is operational
+    - name: Check OpenSearch Running on Linux and MacOS
+      if: ${{ inputs.platform != 'windows-latest'}}
+      run: curl https://localhost:9200 -u 'admin:admin' -k -v
+      shell: bash
+
+    - name: Check OpenSearch Running on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: |
+        $credentialBytes = [Text.Encoding]::ASCII.GetBytes("admin:admin")
+        $encodedCredentials = [Convert]::ToBase64String($credentialBytes)
+        $baseCredentials = "Basic $encodedCredentials"
+        $Headers = @{ Authorization = $baseCredentials }
+        Invoke-WebRequest -SkipCertificateCheck -Uri 'https://localhost:9200' -Headers $Headers;
```

Comment:
Can we change both url to ~`https://admin:admin@localhost:9200/_cat/plugins`


#### `_cat/plugins` output

This is much better at confirming that plugin installed correctly

```
>curl https://admin:admin@localhost:9200/_cat/plugins -k
opensearch-node1-2-2-0 opensearch-alerting                  2.2.0.0
opensearch-node1-2-2-0 opensearch-anomaly-detection         2.2.0.0
opensearch-node1-2-2-0 opensearch-asynchronous-search       2.2.0.0
opensearch-node1-2-2-0 opensearch-cross-cluster-replication 2.2.0.0
opensearch-node1-2-2-0 opensearch-geospatial                2.2.0.0
opensearch-node1-2-2-0 opensearch-index-management          2.2.0.0
opensearch-node1-2-2-0 opensearch-job-scheduler             2.2.0.0
opensearch-node1-2-2-0 opensearch-knn                       2.2.0.0
opensearch-node1-2-2-0 opensearch-ml                        2.2.0.0
opensearch-node1-2-2-0 opensearch-notifications             2.2.0.0
opensearch-node1-2-2-0 opensearch-notifications-core        2.2.0.0
opensearch-node1-2-2-0 opensearch-observability             2.2.0.0
opensearch-node1-2-2-0 opensearch-performance-analyzer      2.2.0.0
opensearch-node1-2-2-0 opensearch-reports-scheduler         2.2.0.0
opensearch-node1-2-2-0 opensearch-security                  2.2.0.0
opensearch-node1-2-2-0 opensearch-sql                       2.2.0.0
```